### PR TITLE
Fixup: restore Content-Type in CloudTasks

### PR DESCRIPTION
**Pull request by Hexcles**
Follow up to #2153 and part of #1747 .

TODO (in a future PR): figure out a way to test this locally. The new SDK no longer provides a local emulator for TaskQueue.

### CODE REVIEW

File: shared/appengine.go [hunk 1]
```diff
@@ -273,7 +273,11 @@ func (a appEngineAPIImpl) ScheduleTask(queueName, taskName, target string, param
 				AppEngineHttpRequest: &taskspb.AppEngineHttpRequest{
 					HttpMethod:  taskspb.HttpMethod_POST,
 					RelativeUri: target,
-					Body:        []byte(params.Encode()),
+					// In appengine.taskqueue, The default for POST task was
+					// application/x-www-form-urlencoded, but the new SDK
+					// defaults to application/octet-stream.
+					Headers: map[string]string{"Content-Tye": "application/x-www-form-urlencoded"},
```
Comment (KyleJu) [inline shared/appengine.go:279]:
```suggestion
					Headers: map[string]string{"Content-Type": "application/x-www-form-urlencoded"},
```

Comment (Hexcles) [reply shared/appengine.go:279]: Oops... Thanks for catching this :)

**Review by KyleJu — APPROVED** [review]
Normally the header format is `map[string][]string`. Kind of weird that its `map[string]string` here. LGTM

File: shared/appengine.go (hunk 1 above)
Comment (KyleJu) [reply shared/appengine.go]: np!

**Pull request merged**